applications/nrf_desktop/src/util/dfu_flash_lock.c

```diff
@@ -0,0 +1,32 @@
+/*
+ * Copyright (c) 2020 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-BSD-5-Clause-Nordic
+ */
+
+
+#include "dfu_flash_lock.h"
+#define FLASH_UNLOCKED	0
+
+
+static atomic_t dfu_flash_locked = ATOMIC_INIT(FLASH_UNLOCKED);
+
+atomic_val_t dfu_flash_get_key(void)
```

Comment:
The api here is extremely misleading. I had to spent few minutes thinking why do you get this key and what is this key for. The other thing is that this key is only for debugging so I am not 100% sure it should be used anyway.

Also you should remove flash from the lock/unlock function names.

it would be much easier to follow if it was
```
typdef atomic_val_t dfu_module_id;

dfu_module_id get_dfu_module_id(void);
bool dfu_lock(dfu_module_id id);
void dfu_unlock(dfu_module_id id);
```

but notice one thing - we already have application module ids which can be used for this purpose.

Reply:
It was done to make sure that module will be able to unlock flash only in case it locked it before. Using module id here is a good idea.